packages/ember/src/components/form/select.gts

```diff
@@ -0,0 +1,433 @@
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { get, action } from '@ember/object';
+import Component from '@glimmer/component';
+// @ts-expect-error Glimmer doesn't currently ship a type for the `cached` decorator
+// https://github.com/glimmerjs/glimmer.js/issues/408
+import { tracked, cached } from '@glimmer/tracking';
+// @ts-expect-error Ember keyboard doesn't currently ship a type for the `on-key` modifier
+import onKey from 'ember-keyboard/modifiers/on-key';
+import { runTask } from 'ember-lifeline';
+
+import BoundValue from './bound-value.ts';
+import onInsert from '../../modifiers/did-insert.ts';
+
+import type { Optional } from '../../types.d.ts';
+
+declare type SelectOption<T> = {
+  label: string;
+  value: string | T;
+  raw: T;
+};
+
+interface SelectItemSignature<T> {
+  Args: {
+    optionIndex: number;
+    activeIndex: number;
+    currentValue: string | T;
+    option: SelectOption<T>;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLLIElement;
+}
+
+class SelectItem<T> extends Component<SelectItemSignature<T>> {
+  get classList() {
+    let classes = ['dropdown-item'];
+
+    const useIndexActive = this.args.activeIndex != -1;
+    const isCurrentIndex = this.args.optionIndex === this.args.activeIndex;
+    const isCurrentValue = this.args.option.value === this.args.currentValue;
+
+    if (useIndexActive && isCurrentIndex) {
+      classes.push('active');
+    }
+
+    if (!useIndexActive && isCurrentValue) {
+      classes.push('active');
+    }
+
+    return classes.join(' ');
+  }
+
+  get isActive() {
+    return this.args.option.value === this.args.currentValue;
```

Comment:
Can you use Ember's [`isEqual`](https://api.emberjs.com/ember/5.10/functions/@ember%2Futils/isEqual) utility for this? If we adopt this pattern, it might help us eliminate the need to "hash" items to compare their equality when dealing with object references.